src/foundry/client/pixi/layers/base/canvas-layer.d.ts

```diff
@@ -0,0 +1,86 @@
+export {};
+
+declare global {
+  /**
+   * An abstract pattern for primary layers of the game canvas to implement
+   * @typeParam Options - The type of the options in this layer.
+   */
+  abstract class CanvasLayer extends PIXI.Container {
+    constructor();
+
+    /**
+     * Options for this layer instance.
+     * @defaultValue `this.constructor.layerOptions`
+     */
+    options: CanvasLayer.LayerOptions;
+
+    /**
+     * @defaultValue `false`
+     */
+    interactiveChildren: boolean;
+
+    /**
+     * Customize behaviors of this CanvasLayer by modifying some behaviors at a class level.
+     */
+    static get layerOptions(): CanvasLayer.LayerOptions;
+
+    /**
+     * The canonical name of the CanvasLayer is the name of the constructor that is the immediate child of the
+     * defined baseClass for the layer type.
+     *
+     * @example
+     * canvas.lighting.name -\> "LightingLayer"
+     * canvas.grid.name -\> "GridLayer"
+     */
+    static get instance(): CanvasLayer | undefined;
+
+    /**
+     * The canonical name of the CanvasLayer
+     * @remarks Foundry defines this as a getter, but since CanvasLayer extends PIXI.Container, it has to be a property.
+     */
+    readonly name: string;
```

Comment:
If I'm correct, above, I honestly have no idea how best to type this.
```js
  get name() {
    const baseCls = this.constructor.layerOptions.baseClass;
    let cls = Object.getPrototypeOf(this.constructor);
    let name = this.constructor.name;
    while ( cls ) {
      if ( cls !== baseCls ) {
        name = cls.name;
        cls = Object.getPrototypeOf(cls);
      }
      else break;
    }
    return name;
  }
```

This seems like an... interesting property.